app/src/components/main.tsx

```diff
@@ -56,14 +62,16 @@ function onFileInputChange(event: React.FormEvent<HTMLInputElement>) {
 
   fileArray.forEach(file => {
     let fr = new FileReader();
-    fr.onload = function () {
-      let result = fr.result as ArrayBuffer
-      var data = new Uint8Array(result);
-
-      Module['FS_createDataFile']('/input/', file.name, data, true, true, true);
-      inputDirectory.next(inputDirectory.getValue().add(file.name))
+    fr.onload = () => {
+      let result = fr.result as ArrayBuffer;
+      const filepath = '/input/' + file.name;
+
+      const subscription = sendFileTransfer(result, filepath).subscribe(message => {
+        const filename: string = message.data.result;
+        inputDirectory.next(inputDirectory.getValue().add(filename));
```

Comment:
Send a file transfer, and then when a response is returned update the input directory widget.